projects/packages/my-jetpack/_inc/components/product-interstitial/index.jsx

```diff
@@ -59,7 +59,10 @@ export default function ProductInterstitial( {
 		recordEvent( 'jetpack_myjetpack_product_interstitial_add_link_click', { product: bundle } );
 	}, [ recordEvent, bundle ] );
 
-	const { isUserConnected } = useMyJetpackConnection();
+	// Search exception 1/2: Search checkout flow won't work with the unlinked=1 approach.
+	const from = 'search' === slug ? 'jetpack-search' : 'my-jetpack';
+
+	const { isUserConnected, handleConnectUser } = useMyJetpackConnection( { from } );
```

Comment:
this expection belongs sto the `useMyJetpackConnection()` hook, so it should probably be moved there.